core/flyout_base.js

```diff
@@ -520,6 +529,66 @@ Blockly.Flyout.prototype.show = function(xmlList) {
 
   this.reflowWrapper_ = this.reflow.bind(this);
   this.workspace_.addChangeListener(this.reflowWrapper_);
+
+  this.recordCategoryScrollPositions_();
+};
+
+/**
+ * Store an array of with category names and scrollbar positions.
+ * This is used when scrolling the flyout to cause a category to be selected.
+ * @private
+ */
+Blockly.Flyout.prototype.recordCategoryScrollPositions_ = function() {
+  this.categoryScrollPositions = [];
+  for (var i = 0; i < this.buttons_.length; i++) {
+    if (this.buttons_[i].isCategoryLabel_) {
+      var categoryLabel = this.buttons_[i];
+      this.categoryScrollPositions.push({
+        categoryName: categoryLabel.text_,
+        position: this.horizontalLayout_ ?
+          categoryLabel.position_.x : categoryLabel.position_.y
+      });
+    }
+  }
+};
+
+/**
+ * Select a category using the scroll position.
+ * @param  {number} pos The scroll position.
```

Comment:
What are the units for pos?

If you're not sure, add a TODO about figuring out the units, so I can follow up on that.

`scaledPos` below seems wrong, though.  Usually `scaledPos` would be a position multiplied by the workspace scale, and this is the inverse.  It's probably the position in workspace units, with pos originally being in pixels, but you should check.